core/src/main/java/lucee/commons/io/compress/CompressUtil.java

```diff
@@ -456,40 +506,51 @@ public static void compressTGZ(Resource[] sources, Resource target, int mode) th
     }
 
     public static void compress7z(Resource[] sources, Resource target) throws IOException {
-     
-     	try (SevenZOutputFile out = new SevenZOutputFile(new File(target.getName()))){
-            for (int i = 0; i < sources.length; i++){
-                compress7zFormat(out, new File(sources[i].getName()), ".");
-            }
-        }
+    	// TODO test arguments
+		RefBoolean isTempFile = new RefBooleanImpl();
+		File trgtFile = toFile(target, isTempFile);
+     	SevenZOutputFile out = new SevenZOutputFile(trgtFile);
+     	for (int i = 0; i < sources.length; i++) {
+     		compress7zFormat(out, sources[i], target);
+     	}
     }
 
 
-     private static void compress7zFormat(SevenZOutputFile out, File file, String dir) throws IOException {
-     	InputStream is = null;
-        String name = dir + File.separator + file.getName();
-        if (file.isFile()){
-            SevenZArchiveEntry entry = out.createArchiveEntry(file, name);
-            out.putArchiveEntry(entry);
-
-            FileInputStream in = new FileInputStream(file);
-            byte[] b = new byte[1024];
-            int count = 0;
-            while ((count = in.read(b)) > 0) {
-                out.write(b, 0, count);
-            }
-            out.closeArchiveEntry();
-
-        } else if (file.isDirectory()) {
-            File[] children = file.listFiles();
-            if (children != null){
-                for (int i = 0; i < children.length; i++){
-                    compress7zFormat(out, children[i], name);
-                }
-            }
-        }
+     private static void compress7zFormat(SevenZOutputFile out, Resource file, Resource targetDir) throws IOException {
+		Resource trg;
+		 // write Gzip
+	    InputStream is = null;
+	    OutputStream os = null;
+		RefBoolean isTempFile2 = new RefBooleanImpl();
+		File srcFile = toFile(file, isTempFile2);
+	    is = new FileInputStream(srcFile);
+		try {
+	     	SevenZArchiveEntry entry = out.createArchiveEntry(srcFile, file.getName());
+	        out.putArchiveEntry(entry);
+	        trg = toResource(targetDir, entry);
```

Comment:
this is not really working, at least not on windows, I get a really weird path something like E:\test\a.zip\test.txt, test being the source and a.zip being the destination